CODEBASE_ANALYSIS.md

```
@@ -0,0 +1,193 @@
+```
+┌─────────────────────────────────────────────────────────────────┐
+│                         main.py                                  │
+│                    (Entry point, SelfBot)                        │
+└─────────────────────────────────────────────────────────────────┘
+                              │
+                              ▼
+┌─────────────────────────────────────────────────────────────────┐
+│                    discord_bot/                                  │
+│  ┌──────────────┐  ┌──────────────┐  ┌──────────────┐          │
+│  │ chat_handler │  │   backfill   │  │context_cache │          │
+│  │(main events) │  │(history sync)│  │(prompt build)│          │
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)